nixos/modules/services/misc/gitea.nix

```diff
@@ -454,6 +479,47 @@ in
       ];
     };
 
+    services.nginx = lib.mkIf cfg.configureNginx {
+      enable = true;
+      virtualHosts."${cfg.domain}" = {
+        root = "/var/empty";
+
+        locations."/" = {
+          proxyPass = (if cfg.enableUnixSocket then "http://unix:/run/gitea/gitea.sock" else "http://127.0.0.1:${toString cfg.httpPort}");
```

Comment:
```suggestion
          proxyPass = if cfg.enableUnixSocket then "http://unix:/run/gitea/gitea.sock" else "http://127.0.0.1:${toString cfg.httpPort}";
```

Reply:
One could also use `"http://unix:${cfg.settings.server.HTTP_ADDR}"` instead of 
 `"http://unix:/run/gitea/gitea.sock"`.
 
And if we go down that route, it might also be worth setting `cfg.httpAddress = "127.0.0.1"` (which in turn sets `cfg.settings.server.HTTP_ADDR`) when `cfg.configureNginx` is enabled to further deduplicate the addresses used and allow something like:

```nix
proxyPass = "http://" +
  (lib.optionalString cfg.enableUnixSocket "unix:") +
  cfg.settings.server.HTTP_ADDR +
  (lib.optionalString (!cfg.enableUnixSocket) ":${toString cfg.httpPort}");
```

Or assert `cfg.enableUnixSocket = true` when `cfg.configureNginx = true` instead of just setting the default for it.
That would at least simplify the logic /shrug

Reply:
Update to:
```
proxyPass = if cfg.enableUnixSocket then "http://unix:${cfg.settings.server.HTTP_ADD}" else "http://127.0.0.1:${toString cfg.httpPort}";
```

> And if we go down that route, it might also be worth setting `cfg.httpAddress = "127.0.0.1"` (which in turn sets `cfg.settings.server.HTTP_ADDR`) when `cfg.configureNginx` is enabled to further deduplicate the addresses used and allow something like:

By default `HTTP_ADDR = "0.0.0.0";`, not `127.0.0.1`.

> Or assert `cfg.enableUnixSocket = true` when `cfg.configureNginx = true` instead of just setting the default for it.
> That would at least simplify the logic /shrug

Change to:
```
     enableUnixSocket = mkOption {
        type = types.bool;
        default = cfg.configureNginx;
```
or:
```
     enableUnixSocket = mkOption {
        type = types.bool;
        default = true;
```
?